src/platforms/javascript/common/session-replay/understanding-sessions.mdx

```diff
@@ -93,3 +93,82 @@ await replay.flush();
 ```
 
 In `session` mode, this will upload any pending recording data to Sentry. In `buffer` mode, this will upload any pending recording data to Sentry and then continue recording, the same as when an error is sampled with `replaysOnErrorSampleRate`.
+
+## Examples of Custom Sampling
+
+There are ways to enable custom sampling if you're interested in tracking a particular action or group, for example, a specific segment of users or a problematic URL.
+
+### Replays for Specific Users
+
+You can set up replays to be recorded automatically for a particular user group. Here's an example where we want to record session-based replays for all users who are employees. The `user` object has an `isEmployee` field to signal employee status.
+
+```javascript
+const replay = new Sentry.Replay();
+
+/**
+ * Initialize the Sentry SDK as normal.
+ *
+ * `replaysSessionSampleRate` is set to its default value of "0.1" so that only ~10% of users are sampled
+ * `replaysOnErrorSampleRate` is set to its default value of "1.0" to always record a replay when an error occurs
+ */
+Sentry.init({
+  dsn: "...",
+  replaysSessionSampleRate: 0.1,
+  replaysOnErrorSampleRate: 1.0,
+  integrations: [replay],
+});
+
+// ...
+// Sometime after your users are authenticated, check if they are an employee
+// If they are, we call `replay.flush()` to either flush the replay as normal if session-based replay, or if it is buffering due to error sampling, the flush will send the first segment of the replay.
+
+// Since `replaysOnErrorSampleRate` is 1, we know that a replay has been buffered and `flush()` will flush the contents of that buffer. If both sample rates are < 1, then there is a chance that a replay has not been recording/buffering.
+// In this case, you can check that `replay.getReplayId()` returns a value, if it does, it means replay is active and you can call `replay.flush()`, other call `replay.start()` to start recording a new replay.
+// ...
```

Comment:
How does this sound @mydea ?

Reply:
it's not fully correct - the `replaysOnErrorSampleRate` only needs to be > 0. As long as this is > 0, you'll always have a buffer. So the instructions should be something like:

As long as `replaysOnErrorSampleRate` is greater than 0, we will always buffer the replay in the background. 
When you then call `flush()`, we will ensure to always send whatever has been buffered before.

It is also safe to call `flush()` in any time, really. If the replay is not enabled, it will simply do nothing. If the replay is currently in `session` mode, it will send what has been buffered, and if it is in `buffer` mode, it will also send what has been buffered _and_ convert the session to `session` mode, like when an error occurred.